3.sc_Morphology_Linear_Modeling/scripts/1b.fit_linear_model_2beta.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+import sys
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+import pyarrow as pa
+import pyarrow.parquet as pq
+import seaborn as sns
+from matplotlib import pyplot as plt
+from pycytominer.cyto_utils import infer_cp_features
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import LabelEncoder
+
+sys.path.append("..")
+# from ..utils.utils import df_stats
+import matplotlib.pyplot as plt
+
+pd.set_option("mode.chained_assignment", None)
+
+
+# In[2]:
+
+
+# Parameters
+celltype = "PBMC"
+
+
+# In[3]:
+
+
+# Define inputs
+feature_file = pathlib.Path(f"../data/{celltype}_preprocessed_sc_norm.parquet")
+feature_df = pq.read_table(feature_file).to_pandas()
+
+
+# In[4]:
+
+
+# if path does not exist, create one
+pathlib.Path(f"./results/{celltype}").mkdir(parents=True, exist_ok=True)
+
+# define output file path
+one_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_one_beta.tsv")
+two_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_two_beta.tsv")
+three_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_three_beta.tsv")
+four_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_four_beta.tsv")
+
+
+# In[5]:
+
+
+cp_features = infer_cp_features(feature_df)
+print(f"We are testing {len(cp_features)} features")
+
+new_line = "\n"
+print(
+    f"The unique Treatment-Dosages are: {f', {new_line}'.join((feature_df['twob_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()))}"
+)
+
+
+# #### Complex Linear Modeling (cell count btea + 2 beta approach)
+# Here I run the same analysis as above but with dosage of a treatment being a factor in the linear model. All features and treatments will be exported into 1 file.
+#
+# Linear Model:
+# $y = \beta _{0}x+ \beta _{1}x+ \beta _{2}x+ \epsilon$ where;
+# $y$ is each feature
+# $x$ is the inputed variables
+# $\beta _{0}$ is the beta coefficient attributed to cell count.
+# $\beta _{1}$ is the beta coefficient attributed to Inducer, Inhibitor, and Inhibitor Dose.
+# $\beta _{2}$ is the beta coefficient attributed to Inducer dose.
+# $\epsilon$ is the residual variance not explained by factors in the model
+
+# In[6]:
+
+
+# Loop for each treatment then each feature
+
+# define the control and treatment
+# Setup linear modeling framework
```

Comment:
It make make more sense to reorder these comments. Ex: move `Loop for each treatment then each feature` to right above the `for treatment in features_df...` code.